plugins/destination/mysql/client/schema.go

```diff
@@ -165,7 +165,9 @@ func (c *Client) createTable(ctx context.Context, table *schema.Table) error {
 			sqlType := arrowTypeToMySqlStr(column.Type)
 			if sqlType == "blob" || sqlType == "text" {
 				// `blob/text` SQL types require specifying prefix length to use for the primary key
-				builder.WriteString("(64)")
+				// https://dev.mysql.com/doc/refman/8.0/en/innodb-limits.html
+				// The index key prefix length limit is 767 bytes for InnoDB tables that use the REDUNDANT or COMPACT row format. For example, you might hit this limit with a column prefix index of more than 191 characters on a TEXT or VARCHAR column, assuming a utf8mb4 character set and the maximum of 4 bytes for each character.
+				builder.WriteString("(191)")
```

Comment:
I think this will be a breaking change because it'll require force migration, right?

Reply:
Agreed that in theory it is a breaking change, but in practice I don't think we validate the length of the PK so it wouldn't cause any issues

Reply:
No, because PK length is not the same as the column length... The column is of type `text` or `blob` which can be up to 64,000 characters, but the index will only use the first 191 characters as part of the PK... So when we query to get the table we don't actually get the details about the PK, only the type of the column

Reply:
If we were to validate that the details of the PK were unchanged (ie the length of the prefix being used) then yes this would be a breaking change

Reply:
Ah okay gotcha, nice. I thought this was the column, not the PK. But next question: will this fix only work if you first forcibly drop the table and then recreate it? Or will the PK get updated automatically?

Reply:
The new prefix size (`191` vs `64`)... Will only be used for new tables (or tables that are dropped and then created)

Reply:
Ok, makes sense. Probably just something to note in the release notes 👍